### PR TITLE
Faster synchrobench init time and more informative heap usage information

### PR DESCRIPTION
**Pull request by liran-funaro**
This PR "shaves" up to 40 seconds from the benchmark's initialization (fill) time.
This can speed up the benchmarking process by up to 30%.

It also adds a more informative heap usage information.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: benchmarks/synchrobench/src/main/java/com/oath/oak/synchrobench/contention/benchmark/Test.java [hunk 1]
```diff
@@ -95,25 +95,26 @@ protected synchronized Random initialValue() {
...
+            int v = localRand.nextInt(range);
+
+            MyBuffer key = new MyBuffer(Parameters.keySize);
+            key.buffer.putInt(0, v);
+            MyBuffer val = new MyBuffer(Parameters.valSize);
+            val.buffer.putInt(0, v);
+            if (oakBench.putIfAbsentOak(key, val)) {
+                i--;
             }
+            // counts all the putIfAbsent operations, not only the successful ones
```
[the remark this answers is not in the document]
Comment (liran-funaro) [reply benchmarks/synchrobench/src/main/java/com/oath/oak/synchrobench/contention/benchmark/Test.java:116]: It wasn't removed. See line 117. The modification simply moves the switch/case outside of the loop.

File: benchmarks/synchrobench/src/main/java/com/oath/oak/synchrobench/contention/benchmark/Test.java [hunk 2]
```diff
@@ -228,6 +235,7 @@ private void execute(int milliseconds, boolean maint)
 
         long endTime = System.currentTimeMillis();
         elapsedTime = ((double) (endTime - startTime)) / 1000.0;
+        printHeapStats("After benchmark");
```
[the remark this answers is not in the document]
Comment (liran-funaro) [reply benchmarks/synchrobench/src/main/java/com/oath/oak/synchrobench/contention/benchmark/Test.java:238]: This happens before and after the measured portion of the benchmarks---not during.

[the remark this answers is not in the document]
Comment (sanastas) [reply benchmarks/synchrobench/src/main/java/com/oath/oak/synchrobench/contention/benchmark/Test.java:238]: I meant total time of running benchmark. Not that the measurement was wrong

[the remark this answers is not in the document]
Comment (liran-funaro) [reply benchmarks/synchrobench/src/main/java/com/oath/oak/synchrobench/contention/benchmark/Test.java:238]: That happens only once in an iteration. Its runtime is negligible.

**Pull request merged**